### ISSUE TITLE
Content Security Policy - "unsafe-eval"

### ISSUE THREAD

**Issue by onoufriosm** [epam/ketcher#853]
**Steps to Reproduce**
Add a CSP (in your html entry file) that doesn't allow for "unsafe-eval"

**Expected behavior**
Not throw a CSP error

**Actual behavior**
Throws a CSP error 
```
Refused to evaluate a string as JavaScript because 'unsafe-eval' is not an allowed source of script
```

**Additional context**
I've tracked this down to `ketcher-standalone/src/generate/libindigo.js` line 1230. Specifically `new Function` will cause this. 

Don't have enough context to make a change to this but essentially `new Function` should go away otherwise CSP won't let it run.

**Comment by AllaZhbanova** [comment]
related https://github.com/epam/Indigo/issues/567

**Comment by AndreiMazol** [comment]
Moved to 2.5
cc: @AllaZhbanova

**Comment by martinuncountable** [comment]
@AndreiMazol  @AllaZhbanova

My understanding is that if we introduce an option for ketcher-core/react package to compile the AVJ validation schemas at build time that will remove the `unsafe-eval` dependency.

Is this something being actively worked on?

**Comment by ahmepham** [comment]
@martinuncountable Do you have any update on this.  It is currently blocking my ability to use ketcher in my app, which is a bummer since ketcher is a very nice library

**Comment by martinuncountable** [comment]
> @martinuncountable Do you have any update on this. It is currently blocking my ability to use ketcher in my app, which is a bummer since ketcher is a very nice library

Yes we have forked Ketcher to use `json-schema` instead. We are working on creating a PR here to merge upstream.

**Comment by NicolasCARPi** [comment]
For information, the PR fixing this issue is opened here: https://github.com/epam/ketcher/pull/6269

Let's hope it can be reviewed and merged (relatively) quickly :)
